### PR TITLE
Adding Pattern Support

### LINKED ISSUE #187: Add Pattern Support
**Current Behavior**
No Pattern Support

**Desired Behavior**
Add Pattern Support

---

### Contributor Resources

Refer: https://github.com/meshery/meshery-osm/pull/103

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- See the [Contributing to Meshery.io Website](https://github.com/layer5io/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- See Meshery site designs in this [Figma designfile](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=110%3A1). Fill-in the [Layer5 Community Member Form](https://layer5.io/newcomer) and join the [Layer5 Community Slack](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Adding the Pattern support for the KUMA mesh.

This PR fixes #187 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: kuma/sample_apps.go [hunk 1]
```diff
@@ -21,3 +25,44 @@ func (kuma *Kuma) installSampleApp(del bool, namespace string, templates []adapt
...
+	// updating the label on the namespace
+	ns, err := kclient.CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if ns.ObjectMeta.Labels == nil {
+		ns.ObjectMeta.Labels = map[string]string{}
+	}
+	ns.ObjectMeta.Labels["openservicemesh.io/monitored-by"] = "kuma"
```
Comment (utkarsh-pro) [inline kuma/sample_apps.go:44]: @sayantan1413 sidecar proxy injection in Kuma Mesh is managed by setting annotation:

```yaml
annotations:
    kuma.io/sidecar-injection: enabled
```

File: templates/oam/workloads/kumamesh.meshery.layer5io.schema.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "$id": "http://meshery.layer5.io/definition/Workload/KUMAMesh",
```
Comment (utkarsh-pro) [inline templates/oam/workloads/kumamesh.meshery.layer5io.schema.json:2]:
```suggestion
    "$id": "http://meshery.layer5.io/definition/Workload/KumaMesh",
```

**Review by utkarsh-pro — CHANGES_REQUESTED** [review]

File: kuma/sample_apps.go (hunk 1 above)
Comment (sayantan1413) [reply kuma/sample_apps.go]: Hey @utkarsh-pro I think this is there in line number 54.

Comment (utkarsh-pro) [reply kuma/sample_apps.go]: @sayantan1413 I meant to say that to enable sidecar injection in Kuma mesh we do not need this lable rather the annotation that I have mentioned above. Also annotation on line 54 is for osm.

Comment (sayantan1413) [reply kuma/sample_apps.go]: Oh okay so i removed the labels now and updated the link to kuma.io😃

File: kuma/sample_apps.go [hunk 3]
```diff
@@ -21,3 +25,35 @@ func (kuma *Kuma) installSampleApp(del bool, namespace string, templates []adapt
...
+	if ns.ObjectMeta.Annotations == nil {
+		ns.ObjectMeta.Annotations = map[string]string{}
+	}
+	ns.ObjectMeta.Annotations["kuma.io/sidecar-injection"] = "enabled"
+
+	if del {
+		delete(ns.ObjectMeta.Annotations, "kuma.io/sidecar-injection")
+	}
+
+	fmt.Println(ns.ObjectMeta)
```
Comment (utkarsh-pro) [inline kuma/sample_apps.go:51]: I think we can remove this log?

File: Dockerfile [hunk 4]
```diff
@@ -22,5 +22,7 @@ FROM gcr.io/distroless/base
 WORKDIR /
 ENV DISTRO="debian"
 ENV GOARCH="amd64"
+ENV SERVICE_ADDR="meshery-kuma"
+ENV MESHERY_SERVER="http://meshery:9081"
```
Comment (utkarsh-pro) [inline Dockerfile:26]: Please add: `COPY templates/ ./templates`

File: kuma/sample_apps.go (hunk 1 above)
Comment (utkarsh-pro) [reply kuma/sample_apps.go]: Thanks

File: kuma/sample_apps.go (hunk 3 above)
Comment (sayantan1413) [reply kuma/sample_apps.go:51]: Yeah sure I am removing this `fmt.Println(ns.ObjectMeta)`

File: Dockerfile (hunk 4 above)
Comment (sayantan1413) [reply Dockerfile:26]: Yeah added it too✔️

**Review by utkarsh-pro — APPROVED** [review]
LGTM. This PR adds Pattern support in the adapter however the adapter will face issue performing concurrent operations due to its thread unsafe nature at the moment. We can track this issue seperately.

**Comment by sayantan1413** [comment]
Yeah sure @utkarsh-pro . Thanks a lot for the help 😃

**Comment by utkarsh-pro** [comment]
Tracking thread unsafe op issue here: https://github.com/meshery/meshery-kuma/issues/191

**Pull request merged**